src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -335,7 +335,14 @@ rpmostree_sysroot_upgrader_new (OstreeSysroot              *sysroot,
 RpmOstreeOrigin *
 rpmostree_sysroot_upgrader_dup_origin (RpmOstreeSysrootUpgrader *self)
 {
-  return rpmostree_origin_dup (self->origin);
+  g_autoptr(RpmOstreeOrigin) ret = rpmostree_origin_dup (self->origin);
+  /* A bit hacky; here we clean out the live state which is deployment specific.
+   * The callers of this API all want a dup so they can create a new one based
+   * on it. Perhaps we should rename to
+   * rpmostree_sysroot_upgrader_prepare_new_origin()?
+   */
+  rpmostree_origin_set_live_state (ret, NULL, NULL);
```

Comment:
Maybe a better place for this is in `parse_origin_deployment ()`? It feels like since the sysroot upgrader will not use the livefs state, conceptually its own copy shouldn't even have it to begin with. E.g., it should work to instantiate a sysroot upgrader and just set it off to pull & deploy, without actually changing the origin.